python/lib/sift_py/report_templates/service.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import List, Optional, cast
+
+from google.protobuf.field_mask_pb2 import FieldMask
+from sift.report_templates.v1.report_templates_pb2 import (
+    CreateReportTemplateRequest,
+    CreateReportTemplateRequestClientKeys,
+    GetReportTemplateRequest,
+    GetReportTemplateResponse,
+    ReportTemplate,
+    ReportTemplateRule,
+    ReportTemplateTag,
+    UpdateReportTemplateRequest,
+)
+from sift.report_templates.v1.report_templates_pb2_grpc import ReportTemplateServiceStub
+
+from sift_py._internal.time import to_timestamp_nanos, to_timestamp_pb
+from sift_py.grpc.transport import SiftChannel
+from sift_py.ingestion.config.yaml.load import load_report_templates
+from sift_py.report_templates.config import ReportTemplateConfig
+
+
+class ReportTemplateService:
```

Comment:
Could we go pretty aggressive with the doc comments for this class? OPtional for the private method unless you think it'll help other devs

Reply:
Updated! For the private ones I just added a docstring to the `_update` method to try to clarify how I'm doing things there.